### PR TITLE
fix: fix `IndexError` when partioning a pdf with `starting_page_number`

### PR DESCRIPTION
**Pull request by awalker4**
The Issue:

When extracting images from pdfs, we use the metadata page number to index into a list of the images. However, the metadata page number can now be changed via `starting_page_number`. To get the true page index, we need to subtract this value.

Testing:

Run this snippet in a python shell. Before the fix, this throws an IndexError. On this branch, it will return the elements.
```
from unstructured.partition.auto import partition
filename = "example-docs/layout-parser-paper-with-table.pdf"
partition(filename, strategy="hi_res", extract_image_block_types=["Image", "Table"], starting_page_number=20)
```

### CODE REVIEW

File: test_unstructured/partition/pdf_image/test_pdf.py [hunk 1]
```diff
@@ -1223,6 +1223,8 @@ def test_partition_pdf_element_extraction(
         if file_mode == "filename":
             elements = pdf.partition_pdf(
                 filename=filename,
+                # Image extraction shouldn't break by setting this
+                starting_page_number=20,
```
Comment (MthwRobinson) [inline test_unstructured/partition/pdf_image/test_pdf.py:1227]: Could we also add a test that asserts that the resulting page number in `metadata` is correct?

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
Looks good! Just requesting one additional test assertion.

**Review by MthwRobinson — APPROVED** [review]
LGTM

**Review by christinestraub — APPROVED** [review]
LGTM!

**Pull request merged**